apm-agent-core/src/main/java/co/elastic/apm/bci/bytebuddy/MinimumClassFileVersionValidator.java

```diff
@@ -0,0 +1,61 @@
+package co.elastic.apm.bci.bytebuddy;
+
+import net.bytebuddy.ClassFileVersion;
+import net.bytebuddy.asm.AsmVisitorWrapper;
+import net.bytebuddy.description.field.FieldDescription;
+import net.bytebuddy.description.field.FieldList;
+import net.bytebuddy.description.method.MethodList;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.implementation.Implementation;
+import net.bytebuddy.jar.asm.ClassVisitor;
+import net.bytebuddy.pool.TypePool;
+import net.bytebuddy.utility.OpenedClassReader;
+
+public enum MinimumClassFileVersionValidator implements AsmVisitorWrapper {
+
+    INSTANCE;
+
+    private static final ClassFileVersion MINIMUM_CLASS_FILE_VERSION = ClassFileVersion.JAVA_V5;
+
+    @Override
+    public ClassVisitor wrap(TypeDescription instrumentedType, ClassVisitor classVisitor, Implementation.Context implementationContext,
+                             TypePool typePool, FieldList<FieldDescription.InDefinedShape> fields, MethodList<?> methods, int writerFlags, int readerFlags) {
+        return new ClassVisitor(OpenedClassReader.ASM_API, classVisitor) {
+            @Override
+            public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
+                final ClassFileVersion classFileVersion = ClassFileVersion.ofMinorMajor(version);
+                if (!classFileVersion.isAtLeast(MINIMUM_CLASS_FILE_VERSION)) {
+                    throw UnsupportedClassFileVersionException.INSTANCE;
+                }
```

Comment:
This means that the order is correct, meaning that you will avoid applying the `ClassVisitor` coming from the `ForAdvice` transformer whenever version is invalid.